distributed/stealing.py

```diff
@@ -415,4 +425,42 @@ def story(self, *keys):
         return out
 
 
+def _has_restrictions(ts):
+    """Determine whether the given task has restrictions and whether these
+    restrictions are strict.
+    """
+    return not ts.loose_restrictions and (
+        ts.host_restrictions or ts.worker_restrictions or ts.resource_restrictions
+    )
+
+
+def _can_steal(thief, ts, victim):
+    """Determine whether worker ``thief`` can steal task ``ts`` from worker
+    ``victim``.
+    """
+    if not _has_restrictions(ts):
+        return True
```

Comment:
This check makes sense to have semantically, but it's also unnecessary given where `_can_steal` is called.